samples/test/containerAnalysis.test.js

```diff
@@ -0,0 +1,325 @@
+const {assert} = require('chai');
+const cp = require('child_process');
+const uuid = require(`uuid`);
+
+const containerAnalysis = require('@google-cloud/containeranalysis');
+const client = new containerAnalysis.v1beta1.GrafeasV1Beta1Client();
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const uuidVal = uuid.v4();
+const noteId = `test-note-${uuidVal}`;
+const resourceUrl = `gcr.io/test-project/test-image-${uuidVal}`;
+const subscriptionId = `occurrence-subscription-${uuidVal}`;
+const timeoutSeconds = 5;
+
+const {PubSub} = require('@google-cloud/pubsub');
+const pubsub = new PubSub();
+const topicName = 'container-analysis-occurrences-v1beta1';
+const topic = pubsub.topic(topicName);
+
+let projectId;
+let formattedParent;
+let formattedNoteName;
+
+describe('Note tests', function() {
+  before(async function() {
+    // define projectId and related vars
+    projectId = await client.getProjectId();
+    formattedParent = `projects/${projectId}`;
+    formattedNoteName = `projects/${projectId}/notes/${noteId}`;
+  });
+  after(async function() {
+    const [allOccurrences] = await client.listOccurrences({
+      parent: formattedParent,
+      filter: `resourceUrl = "${resourceUrl}"`,
+    });
+
+    allOccurrences.forEach(async occurrence => {
+      await client.deleteOccurrence({name: occurrence.name});
+      console.log(`deleted occurrence ${occurrence.name}`);
+    });
+
+    const [allNotes] = await client.listNotes({
+      parent: formattedParent,
+    });
+
+    allNotes.forEach(async note => {
+      await client.deleteNote({name: note.name});
+      console.log(`deleted note ${note.name}`);
+    });
+  });
+  it('should create a note', function() {
+    const output = execSync(`node createNote.js "${projectId}" "${noteId}"`);
+    assert.match(output, new RegExp(`Note ${formattedNoteName} created.`));
+  });
+
+  it('should get note', function() {
+    const output = execSync(`node getNote.js "${projectId}" "${noteId}"`);
+    assert.match(output, new RegExp(`Note name: ${formattedNoteName}`));
+  });
+
+  it('should create occurrence', function() {
+    const output = execSync(
+      `node createOccurrence.js "${projectId}" "${noteId}" "${projectId}" "${resourceUrl}"`
+    );
+    assert.match(output, new RegExp(`Occurrence created`));
+  });
+
+  // TODO: sometimes fails, should try again if no images are found
```

Comment:
If this is going to flake on us, please add the retry here :)

Reply:
yep, makes sense!